[DOC-8516] Release notes for v22.1.22 (#17654)
cockroachdb/docs: src/current/_includes/releases/v22.1/v22.1.22.md

## v22.1.22

Release Date: August 14, 2023

{% include releases/release-downloads-docker-image.md release=include.release %}

<h3 id="v22-1-22-general-changes">General changes</h3>

- Improved the error message that is logged when a changefeed is triggered by dropping the parent database or a type. The previous error was of the format `value type is not BYTES: UNKNOWN`. [#107943][#107943]

<h3 id="v22-1-22-sql-language-changes">SQL language changes</h3>

- When no data is persisted to SQL statistics tables, such as when no flush has occurred or when flushing is disabled, the endpoint's combined view is shown, which includes in-memory data. [#104052][#104052]

<h3 id="v22-1-22-bug-fixes">Bug fixes</h3>

- Fixed a bug where `SHOW DEFAULT PRIVILEGES` returned no information for a database with uppercase or special characters in its name. [#103954][#103954]

> - Fixed a bug that would result in corruption of [encrypted data at rest on a cluster node]({% link v23.1/encryption.md %}). If a node with this corrupted state was restarted, the node could fail to rejoin the cluster. If multiple nodes encountered this bug at the same time during roll out, the cluster could lose [quorum](https://www.cockroachlabs.com/docs/v23.1/architecture/replication-layer#overview). This note will be updated when more information is available for Technical Advisory 106617.[#104141][#104141]

Author: ```suggestion
- Fixed a bug that would result in corruption of [encrypted data at rest on a cluster node]({% link v23.1/encryption.md %}). If a node with this corrupted state was restarted, the node could fail to rejoin the cluster. If multiple nodes encountered this bug at the same time during roll out, the cluster could lose [quorum]({% link v23.1/architecture/replication-layer.md %}#overview). This note will be updated when more information is available for Technical Advisory 106617.[#104141][#104141]
```

## v22.1.22

Release Date: August 14, 2023

{% include releases/release-downloads-docker-image.md release=include.release %}

<h3 id="v22-1-22-general-changes">General changes</h3>

> - Improved the error message that is logged when a changefeed is triggered by dropping the parent database or a type. The previous error was of the format `value type is not BYTES: UNKNOWN`. [#107943][#107943]

Author: @miretskiy PTAL

Reviewer: Sgtm

<h3 id="v22-1-22-sql-language-changes">SQL language changes</h3>

- When no data is persisted to SQL statistics tables, such as when no flush has occurred or when flushing is disabled, the endpoint's combined view is shown, which includes in-memory data. [#104052][#104052]

<h3 id="v22-1-22-bug-fixes">Bug fixes</h3>

> - Fixed a bug where `SHOW DEFAULT PRIVILEGES` returned no information for a database with uppercase or special characters in its name. [#103954][#103954]

Author: @rafiss PTAL

Reviewer 2: lgtm

> - Fixed a bug where additional empty bytes were written to files in Amazon S3 when the internal cluster setting `cloudstorage.s3.buffer_and_put_uploads.enabled` is `true`. [#104132][#104132]

Author: @stevendanna PTAL

Author: @dt Would you mind taking a look on Steve's behalf since you reviewed the upstream change and it's evening for him?

Reviewer 3: This can be dropped.  This bug never shipped in 22.1 and this is a hidden feature flag.

Author: ```suggestion
```

Author: Done. Thanks!

> - Fixed a bug where running a debug command that manipulates a store, such as `debug compact`, on a running node could corrupt the store if [Encryption At Rest]({% link v23.1/encryption.md %}) was enabled. [#104141][#104141]

Author: ```suggestion
- Fixed a bug that would result in corruption of [encrypted data at rest on a cluster node]({% link v23.1/encryption.md %}). If a node with this corrupted state was restarted, the node could fail to rejoin the cluster. If multiple nodes encountered this bug at the same time during roll out, the cluster could lose [quorum](https://www.cockroachlabs.com/docs/v23.1/architecture/replication-layer#overview). This note will be updated when more information is available for Technical Advisory 106617.[#104141][#104141]
```

> - Fixed a bug where running a debug command that manipulates a store, such as `debug compact`, on a running node could corrupt the store if [Encryption At Rest]({% link ../v23.1/encryption.md %}) was enabled. [#104141][#104141]

Author: @jbowens PTAL

Reviewer 4: While this release note looks good, can we please include a release note for the technical advisory bug fix in this release? Looks like [the commit](https://github.com/cockroachlabs/release-staging/commit/02eef9b3b3e58c9db12d5f64f9a8577ad6ea6ea4) that fixed the bug didn't include a `Release note`, but it should have.

Reviewer 4: Looks like the note for the corresponding bug fix in 23.1.8 was:

Fixed a bug that would result in corruption of [encrypted data at rest on a cluster node](https://www.cockroachlabs.com/docs/v23.1/security-reference/encryption#encryption-at-rest). If a node with this corrupted state was restarted, the node could fail to rejoin the cluster. If multiple nodes encountered this bug at the same time during roll out, the cluster could lose [quorum](https://www.cockroachlabs.com/docs/v23.1/architecture/replication-layer#overview). This note will be updated when more information is available for Technical Advisory 106617. [#107277](https://github.com/cockroachdb/cockroach/pull/107277)

Author: Updated! Thanks for finding that.

> - Fixed a bug where running a debug command that manipulates a store, such as `debug compact`, on a running node could corrupt the store if [Encryption At Rest]({% link /docs/stable/encryption %}) was enabled. [#104141][#104141]

Author: ```suggestion
- Fixed a bug where running a debug command that manipulates a store, such as `debug compact`, on a running node could corrupt the store if [Encryption At Rest]({https://cockroachlabs.com/docs/stable/encryption) was enabled. [#104141][#104141]
```

> - Fixed a null-pointer exception introduced in v22.2.9 and v23.1.1 that could cause a node to crash when populating SQL Activity pages in the DB Console. [#104052][#104052]

Author: @maryliag PTAL

<div class="release-note-contributors" markdown="1">

<h3 id="v22-1-22-contributors">Contributors</h3>

This release includes 7 merged PRs by 9 authors.

</div>

[#103954]: https://github.com/cockroachdb/cockroach/pull/103954
[#104052]: https://github.com/cockroachdb/cockroach/pull/104052

> [#104132]: https://github.com/cockroachdb/cockroach/pull/104132

Author: ```suggestion
```
